cpp/include/cudf/io/json.hpp

```diff
@@ -130,6 +130,18 @@ class json_reader_options {
   // Whether to recover after an invalid JSON line
   json_recovery_mode_t _recovery_mode = json_recovery_mode_t::FAIL;
 
+  // Validation checks for spark
+  // Should the json validation be strict of not
```

Comment:
Please clarify what is "strict" here.

Reply:
@revans2 would you like to keep 3 booleans for strict validation as separate boolean option or a single struct with 3 booleans? (if we want to add more boolean options here then, it may break backward compatibility).

Reply:
Strict to me means that it enforces the JSON specification https://www.json.org/json-en.html. We probably should update the comment to explain that.

As for how we enable and disable things, I am fine with whatever so long as there is a clear way to put it into a mode where it conforms to the JSON spec and that there are ways to enable/disable other individual checks. I am not super concerned about backwards compatibility on the C++ side, as java is more forgiving in that respect.

There are a lot of options here that we could use. If we really want to tie these together we could have a separate builder that is specific for the extra validation steps. That way we can maintain backwards compatibility when/if we add in more validation options, and it conforms with the same pattern used to configure the readers.